cmd/ping.go

```diff
@@ -19,50 +19,47 @@ import (
 	"encoding/json"
 	"fmt"
 	"net/http"
-	"time"
 
 	"github.com/jacktrip/jacktrip-agent/pkg/client"
 )
 
 const (
-	// AgentPingURL is the URL used to POST agent pings
+	// AgentPingURL is the URL used to POST HTTP heartbeats
 	AgentPingURL = "/agents/ping"
 
-	// AgentPingInterval is an interval between pings
-	AgentPingInterval = 5
+	// HeartbeatInterval is an interval between heartbeats
+	HeartbeatInterval = 5
 )
 
-// sendPing sends ping to service to retrieve config
-func sendPing(ping client.AgentPing, apiOrigin string) (client.AgentConfig, error) {
+// sendHTTPHeartbeat sends HTTP heartbeat to api and receives latest config
+func sendHTTPHeartbeat(beat interface{}, apiOrigin string) (client.AgentConfig, error) {
 	var config client.AgentConfig
 
-	ping.StatsUpdatedAt = time.Now()
-
-	// update and encode ping content
-	pingBytes, err := json.Marshal(ping)
+	// update and encode heartbeat content
+	beatBytes, err := json.Marshal(beat)
 	if err != nil {
-		log.Error(err, "Failed to marshal agent ping request")
+		log.Error(err, "Failed to marshal agent heartbeat request")
 		return config, err
 	}
 
-	// send ping request
-	r, err := http.Post(fmt.Sprintf("%s%s", apiOrigin, AgentPingURL), "application/json", bytes.NewReader(pingBytes))
+	// send heartbeat request
+	r, err := http.Post(fmt.Sprintf("%s%s", apiOrigin, AgentPingURL), "application/json", bytes.NewReader(beatBytes))
```

Comment:
can we parameterize AgentPingURL since we'll be using separate endpoints for device and audio server? In the develop branch, there should be `DeviceHeartbeatPath` const in cmd/device.go and `ServerHeartbeatPath` for cmd/server.go

Reply:
Yes. Note that `DeviceHeartbeatPath` already exists in `device.go`. I think more changes (including more refactoring and renaming) will be required to implement the server version of all this, so I was thinking we could defer it to later.